### PR TITLE
Apply remaining changes for supporting "Class" field

### PR DESCRIPTION
**Pull request by Steve-Mcl**
Hi Dave, I have fetched v3, merged it locally and added the remaining changes as discussed in #708  

Here is a demo flow for testing...

WnuKqpCNRE

```
[{"id":"f4b7f5d47f78fcfc","type":"inject","z":"af952aeaa20f4f97","name":"","props":[{"p":"topic","vt":"str"},{"p":"payload"},{"p":"className","v":"notification-error","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"ERROR","payload":"An error happened","payloadType":"str","x":160,"y":260,"wires":[["07e9633b37b26775"]]},{"id":"07e9633b37b26775","type":"ui_toast","z":"af952aeaa20f4f97","position":"top right","displayTime":"50","highlight":"","sendall":true,"outputs":0,"ok":"OK","cancel":"","raw":true,"className":"","topic":"","name":"","x":780,"y":240,"wires":[]},{"id":"887c3c77e479e52e","type":"ui_template","z":"af952aeaa20f4f97","group":"dce9e7a2.d20c78","name":"styles","order":11,"width":0,"height":0,"format":"<style>\n  /* notification colors */\n  md-toast.notification-info {\n    border-width: 10px;\n    border-color: darkgreen;\n  }\n  md-toast.notification-info > h3 {\n    background-color: green;\n  }\n  md-toast.notification-info > div {\n    background: rgba(0, 255, 0, 0.9);\n    color: darkgreen;\n  }\n\n  md-toast.notification-warn {\n    border-width: 10px;\n    border-color: darkorange;\n  }\n  md-toast.notification-warn > h3 {\n    background-color: orange;\n  }\n  md-toast.notification-warn > div {\n    background: rgba(245, 173, 66, 0.9);\n    color: darkorange;\n  }\n  \n  md-toast.notification-error {\n    border-width: 10px;\n    border-color: darkred;\n  }\n  md-toast.notification-error > h3 {\n    background-color: red;\n  }\n  md-toast.notification-error > div {\n    background: rgba(255, 0, 0, 0.9);\n    color: darkred;\n  }\n\n\n  /* Button colors */\n  .nr-dashboard-theme .button-info.nr-dashboard-button button.md-button {\n    background: rgba(0, 255, 0, 0.5);\n    color: rgb(0, 30, 0);\n  }\n  .nr-dashboard-theme .button-error.nr-dashboard-button button.md-button {\n    background: rgba(255, 0, 0, 0.5);\n    color: rgb(30, 0, 0);\n  }\n  .nr-dashboard-theme .button-warn.nr-dashboard-button button.md-button {\n    background: rgba(245, 173, 66, 0.5);\n    color: rgb(145, 73, 22);\n  }\n\n  md-card._md.blue-card {\n      background: lightblue;\n  }\n\n  md-card._md.gray-card {\n    background: lightgray;\n  }\n\n  md-card.yellow-card.visible {\n    background: yellow !important;\n  }\n\n  md-card.visible.bold-span-text > span {\n    font-weight: bold !important;\n  }\n  \n  md-card.visible.text-emphasis {\n    font-style: italic !important;\n    font-weight: 800 !important;\n  }\n\n  md-card.visible.bold-p-label > p.label {\n    font-weight: bold !important;\n  }\n\n  md-card.my-chart._md.layout-align-center-center.visible {\n    background: lightgoldenrodyellow;\n  }\n\n  md-card.my-chart._md.layout-align-center-center.visible\n  p.label.nr-dashboard-chart-title.nr-dashboard-chart-titlel {\n    font-family: cursive;\n    font-size: larger;\n    font-stretch: extra-expanded;\n    font-style: italic;\n    font-variant-caps: small-caps;\n    background-color: aquamarine;\n    border: black;\n    border-width: 2px;\n    border-style: solid;\n    padding-top: 0px;\n  }\n\n\n  .my-group {\n    background: brown !important;\n  }  \n  .my-group p.nr-dashboard-cardtitle {\n    font-size: 30px;\n    font-style: italic;\n    margin: 0px;\n    padding-right: 12px;\n    padding-bottom: 12px !important;\n  }\n  md-card.my-form.nr-dashboard-form {\n    background: peachpuff;\n  }\n  md-card.my-form button[type=submit] {\n    background: green;\n  }\n  md-card.my-form button[type=submit]:hover {\n    background: red !important;\n  }\n  md-card.my-form-dark.nr-dashboard-form {\n    background: rgb(0,0,50);\n  }\n  md-card.my-form-dark button[type=submit] {\n    background: rgb(0,0,105);\n  }\n  md-card.my-form-dark button[type=submit]:hover {\n    background: background: rgb(0,25,75) !important;\n  }\n  \n  /* md-card.nr-dashboard-form._md.visible.my-form-dark > form > p */\n  md-card.nr-dashboard-form.my-form-dark > form > p {\n    color: white;\n  }\n\n  md-card.nr-dashboard-form.my-form-dark > form md-input-container .md-input {\n    color: white;\n  }\n</style>","storeOutMessages":true,"fwdInMessages":true,"resendOnRefresh":true,"templateScope":"global","className":"","x":630,"y":60,"wires":[[]]},{"id":"372dc9715e13dbc4","type":"inject","z":"af952aeaa20f4f97","name":"","props":[{"p":"topic","vt":"str"},{"p":"payload"},{"p":"className","v":"notification-info","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"INFO","payload":"Some info","payloadType":"str","x":120,"y":180,"wires":[["07e9633b37b26775"]]},{"id":"274974bef8ef7e49","type":"inject","z":"af952aeaa20f4f97","name":"","props":[{"p":"topic","vt":"str"},{"p":"payload"},{"p":"className","v":"notification-warn","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"WARN","payload":"This is a warning","payloadType":"str","x":150,"y":220,"wires":[["07e9633b37b26775"]]},{"id":"99cd63bd68e4608a","type":"ui_button","z":"af952aeaa20f4f97","name":"","group":"da148e61.f015f","order":11,"width":"4","height":"1","passthru":false,"label":"show error","tooltip":"","color":"","bgcolor":"","className":"button-error","icon":"","payload":"This is an error","payloadType":"str","topic":"ERROR TITLE","topicType":"str","x":370,"y":340,"wires":[["489d9ee5f15aa662"]]},{"id":"489d9ee5f15aa662","type":"change","z":"af952aeaa20f4f97","name":"","rules":[{"t":"set","p":"className","pt":"msg","to":"notification-error","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":570,"y":340,"wires":[["07e9633b37b26775"]]},{"id":"60b8fccd18b8f5dd","type":"ui_button","z":"af952aeaa20f4f97","name":"","group":"da148e61.f015f","order":11,"width":"4","height":"1","passthru":false,"label":"show info","tooltip":"","color":"","bgcolor":"","className":"button-info","icon":"","payload":"This is just information","payloadType":"str","topic":"INFO TITLE","topicType":"str","x":360,"y":380,"wires":[["c84b76cc56f35b51"]]},{"id":"c84b76cc56f35b51","type":"change","z":"af952aeaa20f4f97","name":"","rules":[{"t":"set","p":"className","pt":"msg","to":"notification-info","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":570,"y":380,"wires":[["07e9633b37b26775"]]},{"id":"aee19488541fe52f","type":"ui_button","z":"af952aeaa20f4f97","name":"","group":"da148e61.f015f","order":11,"width":"4","height":"1","passthru":false,"label":"show warn","tooltip":"","color":"","bgcolor":"","className":"button-warn","icon":"","payload":"This is a warning","payloadType":"str","topic":"WARN TITLE","topicType":"str","x":370,"y":420,"wires":[["16ca42fe5bd2a0eb"]]},{"id":"16ca42fe5bd2a0eb","type":"change","z":"af952aeaa20f4f97","name":"","rules":[{"t":"set","p":"className","pt":"msg","to":"notification-warn","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":570,"y":420,"wires":[["07e9633b37b26775"]]},{"id":"7de4faabaa5cd06f","type":"inject","z":"af952aeaa20f4f97","name":"set class button-error","props":[{"p":"ui_controlX","v":"{ \"className\":\"button-error\" }","vt":"json"},{"p":"topic","vt":"str"},{"p":"className","v":"button-error","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","x":140,"y":440,"wires":[["aee19488541fe52f"]]},{"id":"d6b65ffa68be66c2","type":"inject","z":"af952aeaa20f4f97","name":"set class button-warn","props":[{"p":"ui_controlX","v":"{ \"className\":\"button-warn\" }","vt":"json"},{"p":"topic","vt":"str"},{"p":"className","v":"button-warn","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","x":140,"y":400,"wires":[["aee19488541fe52f"]]},{"id":"ad25abafd6e5d947","type":"ui_switch","z":"af952aeaa20f4f97","name":"","label":"{{className}}","tooltip":"","group":"da148e61.f015f","order":4,"width":0,"height":0,"passthru":false,"decouple":"false","topic":"payload.content","topicType":"msg","style":"","onvalue":"gray-card bold-p-label","onvalueType":"str","onicon":"","oncolor":"","offvalue":"blue-card bold-p-label","offvalueType":"str","officon":"","offcolor":"","animate":false,"className":"gray-card bold-p-label","x":1030,"y":100,"wires":[["e929b5f9670c95e5"]]},{"id":"50d031a9245d8e9c","type":"ui_slider","z":"af952aeaa20f4f97","name":"","label":"slider - class=gray-card","tooltip":"","group":"da148e61.f015f","order":7,"width":0,"height":0,"passthru":true,"outs":"all","topic":"pitch","topicType":"str","min":0,"max":"30","step":"0.5","className":"gray-card","x":1090,"y":260,"wires":[[]]},{"id":"1cbc491f5447379a","type":"ui_slider","z":"af952aeaa20f4f97","name":"","label":"slider - class=blue-card bold-p-label","tooltip":"","group":"da148e61.f015f","order":7,"width":0,"height":0,"passthru":true,"outs":"all","topic":"roll","topicType":"str","min":0,"max":"30","step":"0.5","className":"blue-card bold-p-label","x":1120,"y":320,"wires":[[]]},{"id":"d412219ba3edc8f7","type":"inject","z":"af952aeaa20f4f97","name":"enabled true","props":[{"p":"enabled","v":"true","vt":"bool"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","x":170,"y":500,"wires":[["aee19488541fe52f"]]},{"id":"1f29c7b55eab3db8","type":"inject","z":"af952aeaa20f4f97","name":"enabled false","props":[{"p":"enabled","v":"false","vt":"bool"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","x":170,"y":540,"wires":[["aee19488541fe52f"]]},{"id":"8881611f795336ec","type":"ui_switch","z":"af952aeaa20f4f97","name":"","label":"switch class={{className}}","tooltip":"","group":"da148e61.f015f","order":4,"width":0,"height":0,"passthru":false,"decouple":"false","topic":"payload.content","topicType":"msg","style":"","onvalue":"gray-card","onvalueType":"str","onicon":"","oncolor":"","offvalue":"blue-card","offvalueType":"str","officon":"","offcolor":"","animate":false,"className":"","x":1030,"y":200,"wires":[["9c12a96019addfa3"]]},{"id":"ade3ee9c0252009c","type":"ui_template","z":"af952aeaa20f4f97","group":"da148e61.f015f","name":"yellow-card bold-span-text","order":11,"width":0,"height":0,"format":"This ui-template has the class \"yellow-card bold-span-text\"\n<div>\n    I am the content of a div in a ui-template\n</div>\n\n<pre>\n    I am the content of a pre in a ui-template\n</pre>\n\n<span>\n    I am the content of a span in a ui-template\n</span>\n","storeOutMessages":true,"fwdInMessages":true,"resendOnRefresh":true,"templateScope":"local","className":"yellow-card bold-span-text","x":1090,"y":560,"wires":[[]]},{"id":"dd96667d574643ef","type":"ui_template","z":"af952aeaa20f4f97","group":"da148e61.f015f","name":"yellow-card text-emphasis","order":11,"width":0,"height":0,"format":"This ui-template has the class \"yellow-card text-emphasis\"\n<div>\n    I am the content of a div in a ui-template\n</div>\n\n<pre>\n    I am the content of a pre in a ui-template\n</pre>\n\n<span>\n    I am the content of a span in a ui-template\n</span>\n","storeOutMessages":true,"fwdInMessages":true,"resendOnRefresh":true,"templateScope":"local","className":"yellow-card text-emphasis","x":1090,"y":600,"wires":[[]]},{"id":"491fcaad8d1445a8","type":"ui_chart","z":"af952aeaa20f4f97","name":"","group":"da148e61.f015f","order":9,"width":"12","height":"4","label":"chart without class","chartType":"line","legend":"false","xformat":"HH:mm:ss","interpolate":"linear","nodata":"","dot":false,"ymin":"","ymax":"","removeOlder":"2","removeOlderPoints":"120","removeOlderUnit":"60","cutout":0,"useOneColor":false,"useUTC":false,"colors":["#1f77b4","#aec7e8","#ff7f0e","#2ca02c","#98df8a","#d62728","#ff9896","#9467bd","#c5b0d5"],"outputs":1,"useDifferentColor":false,"className":"","x":1070,"y":380,"wires":[[]]},{"id":"543f1ba81f61d864","type":"inject","z":"af952aeaa20f4f97","name":"","props":[{"p":"payloadxxx","v":"[\t   {\t       \"series\": [\"A\", \"B\", \"C\"],\t       \"data\": [\t           [\t               { \"x\": $toMillis($now()), \"y\": 5 },\t               { \"x\": $toMillis($now())+100, \"y\": 4 },\t               { \"x\": $toMillis($now())+200, \"y\": 2 }     \t           ],\t           [\t               { \"x\": $toMillis($now()), \"y\": 6 },\t               { \"x\": $toMillis($now())+100, \"y\": 7 },\t               { \"x\": $toMillis($now())+200, \"y\": 6 }     \t           ],\t           [\t               { \"x\": $toMillis($now()), \"y\": 7 },\t               { \"x\": $toMillis($now())+100, \"y\": 9 },\t               { \"x\": $toMillis($now())+200, \"y\": 7 }     \t           ] \t       ],\t       \"labels\": [\"\"] \t   }\t]\t","vt":"jsonata"},{"p":"topic","vt":"str"},{"p":"payload"}],"repeat":"2","crontab":"","once":false,"onceDelay":0.1,"topic":"data","payload":"(\t    $minimum := 1;\t    $maximum := 10;\t    $round(($random() * ($maximum-$minimum)) + $minimum, 0)\t)","payloadType":"jsonata","x":890,"y":380,"wires":[["491fcaad8d1445a8","4e7769680e0a7c5f"]]},{"id":"4e7769680e0a7c5f","type":"ui_chart","z":"af952aeaa20f4f97","name":"","group":"da148e61.f015f","order":9,"width":"12","height":"4","label":"chart with class my-chart","chartType":"line","legend":"false","xformat":"HH:mm:ss","interpolate":"linear","nodata":"","dot":false,"ymin":"","ymax":"","removeOlder":"2","removeOlderPoints":"120","removeOlderUnit":"60","cutout":0,"useOneColor":false,"useUTC":false,"colors":["#1f77b4","#aec7e8","#ff7f0e","#2ca02c","#98df8a","#d62728","#ff9896","#9467bd","#c5b0d5"],"outputs":1,"useDifferentColor":false,"className":"my-chart","x":1090,"y":420,"wires":[[]]},{"id":"f0de01d6dd9ef37e","type":"ui_template","z":"af952aeaa20f4f97","group":"6bb8ed49fa5e7238","name":"yellow-card bold-span-text","order":5,"width":0,"height":0,"format":"This ui-template has the class \"yellow-card bold-span-text\"\n<div>\n    I am the content of a div in a ui-template\n</div>\n\n<pre>\n    I am the content of a pre in a ui-template\n</pre>\n\n<span>\n    I am the content of a span in a ui-template\n</span>\n","storeOutMessages":true,"fwdInMessages":true,"resendOnRefresh":true,"templateScope":"local","className":"yellow-card bold-span-text","x":410,"y":540,"wires":[[]]},{"id":"3117438adf7c148f","type":"ui_template","z":"af952aeaa20f4f97","group":"6bb8ed49fa5e7238","name":"yellow-card","order":6,"width":0,"height":0,"format":"This ui-template has the class \"yellow-card\" only\n<div>\n    I am the content of a div in a ui-template\n</div>\n\n<pre>\n    I am the content of a pre in a ui-template\n</pre>\n\n<span>\n    I am the content of a span in a ui-template\n</span>\n","storeOutMessages":true,"fwdInMessages":true,"resendOnRefresh":true,"templateScope":"local","className":"yellow-card","x":370,"y":580,"wires":[[]]},{"id":"71060981eb04c889","type":"ui_button","z":"af952aeaa20f4f97","name":"","group":"6bb8ed49fa5e7238","order":2,"width":"4","height":"1","passthru":false,"label":"↑ Light","tooltip":"","color":"","bgcolor":"","className":"","icon":"","payload":"my-form","payloadType":"str","topic":"topic","topicType":"msg","x":90,"y":60,"wires":[["0b9ffd99e99cdb66"]]},{"id":"a05f07d172940688","type":"ui_button","z":"af952aeaa20f4f97","name":"","group":"6bb8ed49fa5e7238","order":3,"width":"4","height":"1","passthru":false,"label":"↑ Dark","tooltip":"","color":"","bgcolor":"","className":"","icon":"","payload":"my-form-dark","payloadType":"str","topic":"topic","topicType":"msg","x":90,"y":100,"wires":[["0b9ffd99e99cdb66"]]},{"id":"0b9ffd99e99cdb66","type":"change","z":"af952aeaa20f4f97","name":"","rules":[{"t":"set","p":"className","pt":"msg","to":"payload","tot":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":270,"y":60,"wires":[["724f510b2bf64b77"]]},{"id":"724f510b2bf64b77","type":"ui_form","z":"af952aeaa20f4f97","name":"","label":"class=my-form","group":"6bb8ed49fa5e7238","order":1,"width":0,"height":0,"options":[{"label":"Name","value":"Name","type":"text","required":true,"rows":null},{"label":"Age","value":"Age","type":"number","required":false,"rows":null}],"formValue":{"Name":"","Age":""},"payload":"","submit":"submit","cancel":"cancel","topic":"topic","topicType":"msg","splitLayout":"","className":"my-form","x":320,"y":100,"wires":[[]]},{"id":"e929b5f9670c95e5","type":"change","z":"af952aeaa20f4f97","name":"","rules":[{"t":"set","p":"className","pt":"msg","to":"payload","tot":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":1070,"y":60,"wires":[["ad25abafd6e5d947"]]},{"id":"9c12a96019addfa3","type":"change","z":"af952aeaa20f4f97","name":"","rules":[{"t":"set","p":"className","pt":"msg","to":"payload","tot":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":1070,"y":160,"wires":[["8881611f795336ec"]]},{"id":"dce9e7a2.d20c78","type":"ui_group","name":"Object detection","tab":"5132060d.4cde48","order":1,"disp":true,"width":"9","collapse":false},{"id":"da148e61.f015f","type":"ui_group","name":"Demo class=my-group","tab":"da5706ae.6caf58","order":3,"disp":true,"width":"12","collapse":false,"className":"my-group"},{"id":"6bb8ed49fa5e7238","type":"ui_group","name":"another group (no class)","tab":"da5706ae.6caf58","order":2,"disp":true,"width":"8","collapse":false,"className":""},{"id":"5132060d.4cde48","type":"ui_tab","name":"Home","icon":"dashboard","disabled":false,"hidden":false},{"id":"da5706ae.6caf58","type":"ui_tab","name":"Class Demo","icon":"dashboard","disabled":false,"hidden":false}]
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,6 +49,9 @@
         },
         {
             "name": "@hotNipi"
+        },
+        {
+            "name": "@Steve-Mcl"
```
Comment (dceejay) [inline package.json:54]: thanks

**Comment by twocolors** [comment]
@Steve-Mcl thx you, i wait this from https://discourse.nodered.org/t/allow-msg-class-to-add-md-card-to-display-in-widget/40640/6

**Comment by Steve-Mcl** [comment]
@twocolors you know this is now published in npm?

**Comment by twocolors** [comment]
@Steve-Mcl yes, thanks a lot

**Pull request merged**